### PR TITLE
For-R Part B pages and unit tests

### PR DESCRIPTION
**Pull request by babul3012**

### CODE REVIEW

File: src/services/FormsService.ts [hunk 1]
```diff
@@ -16,4 +17,14 @@ export class FormRPartAService extends ApiService {
   async getTraineeFormRPartA(): Promise<AxiosResponse<FormRPartA[]>> {
     return this.get<FormRPartA[]>(`/formr-parta/${this.traineeTisId}`);
   }
+
+  async saveTraineeFormRPartB(
+    formData: FormRPartB
```
Comment (omarCreativeDev) [inline src/services/FormsService.ts:22]: just for my own understanding, is there a reason why promise is being used over an observable?

**Comment by StevenHee** [comment]
Hey Babul.  Just wondered if form validation will be added as part of a separate pr?

File: src/services/FormsService.ts (hunk 1 above)
Comment (john12321) [reply src/services/FormsService.ts:22]: @omarCreativeDev  Would you need to bring in another library such as redux-observable to do this? Not sure what is gained by going down the rxjs route.

**Comment by babul3012** [comment]
> Hey Babul. Just wondered if form validation will be added as part of a separate pr?

ValidationSchema handles the form fields validation. What do you mean by form validation in particular?

File: src/components/forms/formr-part-b/ProfileToFormRPartBInitialValuesMapping.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    forename: pd?.forenames || "",
+    surname: pd?.surname || "",
+    gmcNumber: pd?.gmcNumber || "",
+    email: pd?.email || "",
+    localOfficeName: pd?.personOwner || "",
+    prevRevalBody: pd?.prevRevalBody || "",
+    currRevalDate: pd?.currRevalDate,
+    prevRevalDate: pd?.prevRevalDate,
+    programmeSpecialty: curriculum?.curriculumName || "",
+    dualSpecialty: curriculum?.curriculumName || "",
```
Comment (DorisWongNHS) [inline src/components/forms/formr-part-b/ProfileToFormRPartBInitialValuesMapping.ts:38]: Just had a look at the field validation, seems that the fields 'email' and 'dualSpecialty' can leave unpopulated
https://hee-tis.atlassian.net/wiki/spaces/NTCS/pages/73662481/Trainee+UI+-+Field+Validation+Form+R+Part+B

Comment (StevenHee) [reply src/components/forms/formr-part-b/ProfileToFormRPartBInitialValuesMapping.ts:38]: Basically the same as Doris plus the Date of Previous Revalidation can be in the future.

File: src/services/FormsService.ts (hunk 1 above)
Comment (omarCreativeDev) [reply src/services/FormsService.ts:22]: @john12321 to be honest my react knowledge is quite limited but I assumed observables comes out of the box so im not entirely sure if another library would be needed : )

i do feel tho that observables provide a lot more out of the box but promises are of course also great

File: src/components/forms/formr-part-b/ProfileToFormRPartBInitialValuesMapping.ts (hunk 2 above)
Comment (babul3012) [reply src/components/forms/formr-part-b/ProfileToFormRPartBInitialValuesMapping.ts:38]: @StevenHee Not sure of the date validations atm, will have to check with Ash.

**Comment by StevenHee** [comment]
Does the email field in part b map to the email in traineeProfile.personalDetails or is this separate. Currently showing as blank for new form.

**Comment by john12321** [comment]
@StevenHee [https://hee-tis.atlassian.net/wiki/spaces/NTCS/pages/73662481/Trainee+UI+-+Field+Validation+Form+R+Part+B ](url) shows you you the requirements for the fields in form-R part B. (I think email field should be intentionally blank).

**Pull request merged**